src/H5.c

```diff
@@ -949,7 +966,51 @@ H5check_version(unsigned majnum, unsigned minnum, unsigned relnum)
                 break;
         } /* end switch */
 
-    } /* end if */
+    } /* end if (H5_VERS_MAJOR != majnum || H5_VERS_MINOR != minnum) */
+
+    /* H5_VERS_RELEASE should be compatible, we will only add checks for exceptions */
```

Comment:
This is a copy/paste from above with the wording and checks for ".release" version values.